src/core/server/http/http_server.test.ts

```diff
@@ -992,6 +992,226 @@ describe('body options', () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.delete(
+      {
+        path: '/',
+        validate: false,
+        options: { timeout: 300000 },
+      },
+      (context, req, res) => {
+        try {
+          return res.ok({ body: { timeout: req.route.options.timeout } });
+        } catch (err) {
+          return res.internalError({ body: err.message });
+        }
+      }
+    );
+    registerRouter(router);
+    await server.start();
+    await supertest(innerServer.listener).delete('/').expect(200, {
+      timeout: 300000,
+    });
+  });
+
+  test('should accept a socket "timeout" which is 3 minutes in milliseconds, "300000" for a PUT', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.put(
+      {
+        path: '/',
+        validate: false,
+        options: { timeout: 300000 },
+      },
+      (context, req, res) => {
+        try {
+          return res.ok({ body: { timeout: req.route.options.timeout } });
+        } catch (err) {
+          return res.internalError({ body: err.message });
+        }
+      }
+    );
+    registerRouter(router);
+    await server.start();
+    await supertest(innerServer.listener).put('/').expect(200, {
+      timeout: 300000,
+    });
+  });
+
+  test('should accept a socket "timeout" which is 3 minutes in milliseconds, "300000" for a PATCH', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.patch(
+      {
+        path: '/',
+        validate: false,
+        options: { timeout: 300000 },
+      },
+      (context, req, res) => {
+        try {
+          return res.ok({ body: { timeout: req.route.options.timeout } });
+        } catch (err) {
+          return res.internalError({ body: err.message });
+        }
+      }
+    );
+    registerRouter(router);
+    await server.start();
+    await supertest(innerServer.listener).patch('/').expect(200, {
+      timeout: 300000,
+    });
+  });
+
+  test('should reject a socket "timeout" of "0" for a POST', async () => {
+    const { registerRouter } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.post(
+      {
+        path: '/',
+        validate: false,
+        options: { timeout: 0 },
+      },
+      (context, req, res) => {
+        try {
+          return res.ok({ body: { timeout: req.route.options.timeout } });
+        } catch (err) {
+          return res.internalError({ body: err.message });
+        }
+      }
+    );
+    registerRouter(router);
+    await expect(server.start()).rejects.toThrowError('Invalid routeConfig options');
```

Comment:
1. The message doesn't contain an invalid option name
2. you cover this in http config test https://github.com/elastic/kibana/pull/73103/files#diff-a030b4d8ada02fb446eccfaa965fb895, so probably it's fine not to test this logic here.

Reply:
The test and error is actually me testing and exposing the errors that Hapi server is giving back to us here. I don't do a throw on the zero and neither does the Kibana Schema for this particular test. This test exercises the route as if the route was trying to send down a socket timeout of zero with no Kibana Schema validation and the error that comes back is from Hapi Server which does not tell you which invalid route config option you have. Hapi just does a generic throw which is, `Invalid routeConfig options` and is not really under my control unless I begin catching and abstracting away Hapi exceptions (which I didn't see anywhere else at this moment us doing).

This boundary test I found useful as it gave me confidence that if someone tries to create a route without any schema validation and pushes down a zero then at least Hapi will throw an error and that route will not be considered valid.

Reply:
I'd avoid testing Hapi internals then. It's easy to have false positives on the next Hapi update.

Reply:
Ok, I will remove it.